gluecodium/src/main/resources/templates/cpp/Optional.mustache

```diff
@@ -65,25 +65,16 @@
 #endif
 
 #if !defined(GLUECODIUM_OPTIONAL_HEADER) && !defined(GLUECODIUM_OPTIONAL_TYPE)
-#ifdef __has_include
-# if defined(_MSC_VER)
-#   if _MSC_VER >= 1911 && _MSVC_LANG >= 201703
-#     define GLUECODIUM_OPTIONAL_HEADER <optional>
-#     define GLUECODIUM_OPTIONAL_TYPE std::optional
-#     define GLUECODIUM_OPTIONAL_HELPER_TYPES auto nullopt = std::nullopt; \
-                                          auto in_place = std::in_place; \
-                                          auto bad_access = std::bad_optional_access;
+#  if (defined(_MSC_VER) && _MSC_VER >= 1911 && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L
+#    define GLUECODIUM_OPTIONAL_HEADER <optional>
+#    define GLUECODIUM_OPTIONAL_TYPE std::optional
+#    define GLUECODIUM_OPTIONAL_HELPER_TYPES using std::nullopt; \
+                                             using std::in_place; \
+                                             using std::bad_optional_access;
+#  else
+#    define GLUECODIUM_OPTIONAL_HEADER "OptionalImpl.h"
```

Comment:
Why do we have this here if we have another check in l81

Reply:
Good point, done (removed the version in line 81).